File: go/stack.go
package main

import (
	"fmt"
)

type Stack struct {
	stackData []interface{}
}

func (myStack *Stack) Push(value interface{}) {
	myStack.stackData = append(myStack.stackData, value)
}

func (myStack *Stack) Pop() interface{} {
	stackLength := len(myStack.stackData)
	topValue := myStack.stackData[stackLength-1]
	myStack.stackData = myStack.stackData[:stackLength-1]
	return topValue
}

func (myStack Stack) Top() interface{} {
	stackLength := len(myStack.stackData)
	return myStack.stackData[stackLength-1]
}

func (myStack Stack) Len() int {
	return len(myStack.stackData)
}

func (myStack Stack) IsEmpty() bool {
	return len(myStack.stackData) == 0
}

func (myStack Stack) Print() {
	fmt.Println(myStack.stackData)
}

func main() {
	myStack := &Stack{}
	myStack.Push(5)
	myStack.Push(10)
	myStack.Print()
	fmt.Println(myStack.Len())
	fmt.Println(myStack.Pop())
	fmt.Print(myStack.Top(), " ")
	fmt.Println(myStack.Top())
	fmt.Println(myStack.IsEmpty())
}

File: java/Stack.java
class Stack {
    private int[] data;
    private int size;
    private int top = -1;

    public Stack() {
        this(10);
    }

    public Stack(int size) {
        this.size = size;
        data = new int[size];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public int size() {
        return top + 1;
    }

    public boolean push(int value) {
        if (top +1 == size) {
            System.out.println("Stack overflow!");
            return false;
        } else {
            data[++top] = value;
            return true;
        }
    }

    public int pop() {
        if(!isEmpty()) {
            int topValue = data[top];
            data[top--] = 0;
            return topValue;
        }
        return 0;
    }

    public int top() {
        if (!isEmpty()) {
            return data[top];
        }
        return 0;
    }

    public void print() {
        System.out.println("Current stack: ");
        for (int i = 0; i <= top; i++) {
            System.out.println(data[i]+",");
        }
        System.out.println();
    }

    public static void main(String...args) {
        Stack myStack = new Stack();
        myStack.push(5);
        myStack.push(10);
        System.out.println("Popped: "+myStack.pop());
        myStack.print();
    }
}